app/models/component.rb

```diff
@@ -170,8 +231,12 @@ def from_mapping(srg)
 
   def largest_rule_id
     # rule_id is a string, convert it to a number and then extract the current highest number.
-    Rule.connection.execute("SELECT MAX(TO_NUMBER(rule_id, '999999')) FROM base_rules
-                             WHERE component_id = #{id}")&.values&.flatten&.first&.to_i || 0
+    if id.nil?
+      rules.collect { |rule| rule.rule_id.to_i }.max
```

Comment:
This is probably not very significant, but would it be better in that it reduces memory usage and puts more of the work on the db instead?
```suggestion
      rules.order(:rule_id).pluck(:rule_id).last.to_i
```

Reply:
Does this still work when the rules aren't actually populated in the database yet? The issue this was attempting to correct was that `last_rule_id` was having problems during the initial save when everything was in memory.